### PR TITLE
Minor kubebuilder related items to clean up 

### PR DESCRIPTION
**Pull request by carlisia**
This closes https://github.com/vmware-tanzu/velero/issues/2884.

There is a test file that was deleted bc the functionality was moved to inside the controller, but the controller test is testing the same things anyway (pkg/controller/server_status_request_controller_test.go).

There is no change in logic with this PR.

### CODE REVIEW

File: pkg/cmd/server/server_test.go
Comment (zubron) [inline pkg/cmd/server/server_test.go:144]: Optional, but I think it would be nice if instead of checking the counts, we iterate over the `disabledControllers` slice and check that that key doesn't appear in the `enabledControllers` or `enabledRuntimeControllers` map.

**Review by zubron — APPROVED** [review]
LGTM - thanks! I've got one question, and I've left a suggested change but they aren't blockers.

File: pkg/cmd/server/server_test.go
Comment (carlisia) [reply pkg/cmd/server/server_test.go:144]: I agree. I left the testing of the count too. ✅

File: pkg/controller/server_status_request_controller.go
Comment (carlisia) [reply pkg/controller/server_status_request_controller.go:97]: It's rare that you wouldn't want to re-queue on failure. One case is when the object is not found: most likely it will continue not to be found. In this case, return a nil instead of an error.

Anytime an error is return, the request will be re-queued.

**Review by zubron — DISMISSED** [review]

File: pkg/cmd/server/server_test.go
Comment (zubron) [inline pkg/cmd/server/server_test.go:115]: I'm not sure how to best handle this spelling issue. Would this test still work to have a correctly spelled word, but one that isn't a valid controller name?

Comment (carlisia) [reply pkg/cmd/server/server_test.go:115]: If by invalid controller name you mean a controller that is not on the list of ones that are disabable, then yes, this test would still work the same. You can see it in this other test: https://github.com/vmware-tanzu/velero/pull/3180/files#diff-073f5a2184977287c11f4b4c94aced2fab5718ee4946a2b050d472fa6059496fR105. Included in the list of controllers to disable is the BSL controller, so there'll be an error.

Is this what you mean?

Comment (zubron) [reply pkg/cmd/server/server_test.go:115]: Yeah, I was just thinking of how to trigger the behaviour that we're testing for but fix the codespell issue without adding the word to the list of skipped words as it's something that we could easily misspell elsewhere :)

Comment (carlisia) [reply pkg/cmd/server/server_test.go:115]: Oh duh, I JUST realized what you mean haha. I like this test because it stands for both "a controller that doesn't exist" and "a controller with a very similar name exists but the system handles it properly". Not married to it though. The check is not a merge blocker. I feel like we should have some sorta rule of thumb for things like this.

Comment (carlisia) [reply pkg/cmd/server/server_test.go:115]: Yes, we don't want to add this to the list of skipped words. Let me misspell something else.

**Comment by carlisia** [comment]
This is ready for review.

**Review by jenting — COMMENTED** [review]
I did not see the changes about `remove rbac/role file (config/rbac/role.yaml), as well as it's generation in the script (I think this will work: https://github.com/vmware-tanzu/velero/blob/main/hack/update-generated-crd-code.sh#L50)` in this PR.

**Comment by carlisia** [comment]
> I did not see the changes about remove rbac/role file (config/rbac/role.yaml), as well as it's generation in the script (I think this will work: https://github.com/vmware-tanzu/velero/blob/main/hack/update-generated-crd-code.sh#L50) in this PR.

@jenting I had removed that line and tested it and made no difference. But, I don't think removing it will cause any problem either, so I just removed it.

**Comment by carlisia** [comment]
@vmware-tanzu/velero-maintainers this is ready for review.

File: pkg/controller/server_status_request_controller_test.go
Comment (jenting) [inline pkg/controller/server_status_request_controller_test.go:201]: nit: it would be good if we could calculate the expired time by referencing the `ttl` https://github.com/vmware-tanzu/velero/blob/e69fac1/pkg/controller/server_status_request_controller.go#L37, but I won't block this PR.

**Review by jenting — APPROVED** [review]
LGTM

**Review by zubron — APPROVED** [review]
Thanks!

**Pull request merged**
